### PR TITLE
Energy katana gets deleted when the ninja dies.

### PR DESCRIPTION
**Pull request by PowerfulBacon**
## About The Pull Request

Destroys the energy katana when the ninja dies.
Makes the energy katana indestructible.

## Why It's Good For The Game

Energy katana is one of the most overpowered items in the game, if the ninja manages to die then literally nothing stands a chance against security wielding it.
Also makes it indestructible, so its always in play while the ninja is in play and then out of play when the ninja is out of play.

## Changelog
:cl:
tweak: Energy katana is now indestructible.
tweak: Energy katana is destroyed when its attached ninja is destroyed.
/:cl:

### CODE REVIEW

File: code/modules/ninja/suit/suit.dm [hunk 1]
```diff
@@ -82,7 +82,9 @@ Contents:
 	qdel(n_gloves)
 	qdel(n_shoes)
 	qdel(src)
-
+	if(energyKatana)
+		energyKatana.visible_message("<span class='warning'>[src] flares and then turns to dust!</span>")
+		qdel(energyKatana)
```
Comment (Archanial) [inline code/modules/ninja/suit/suit.dm:86]: Will this work after you qdel ninja himself?

Comment (PowerfulBacon) [reply code/modules/ninja/suit/suit.dm:86]: actually this should definitely go before qdeling src

**Comment by MNarath1** [comment]
that acutally might lead to sec trying to imprison the ninja lol

**Comment by LivingBatteryBox** [comment]
just softcrit them or stamcrit them, cuff, pacify in surgery, then you have a katana with no consequences.

**Comment by PowerfulBacon** [comment]
> 
> 
> just softcrit them or stamcrit them, cuff, pacify in surgery, then you have a katana with no consequences.

The ninja can always blow themselves up. Also only the ninja can choose to remove the suit, so you cannot do surgery on them to remove the explosive implant unless they take the suit off, at which point if they die the suit will be deactivated and wont trigger the self destruct of all the other items.

**Comment by bloons3** [comment]
> The ninja can always blow themselves up.

EMPs will disable their explosive implant

**Pull request merged**